libpod/volume.go

```diff
@@ -283,6 +286,38 @@ func (v *Volume) UsesVolumeDriver() bool {
 	return v.config.Driver != define.VolumeDriverLocal && v.config.Driver != ""
 }
 
+// Pinned returns whether this volume is marked as pinned.
+// Pinned volumes are excluded from system prune operations by default.
+func (v *Volume) Pinned() bool {
+	v.lock.Lock()
+	defer v.lock.Unlock()
+
+	if err := v.update(); err != nil {
+		return false
+	}
+
+	return v.state.Pinned
+}
+
+// SetPinned sets the pinned status of the volume.
+// Pinned volumes are excluded from system prune operations by default.
+func (v *Volume) SetPinned(pinned bool) error {
+	if !v.valid {
+		return define.ErrVolumeRemoved
+	}
+
+	v.lock.Lock()
+	defer v.lock.Unlock()
+
+	if err := v.update(); err != nil {
+		return err
+	}
+
+	v.state.Pinned = pinned
```

Comment:
Should pinning a volume that is already pinned be an error?

Reply:
Made it a no-op. If the volume is already in the desired pinned state, SetPinned() now returns early without error and without calling save(). This follows the principle of idempotency.